airbyte-integrations/bases/bases-destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/AsyncFlush.java

```diff
@@ -94,7 +94,7 @@ public void flush(final StreamDescriptor decs, final Stream<PartialAirbyteMessag
   @Override
   public long getOptimalBatchSizeBytes() {
     // todo(davin): this should be per-destination specific. currently this is for Snowflake.
-    return 100 * 1024 * 1024;
+    return 50 * 1024 * 1024;
```

Comment:
Any comments to where this magic number came from?

Reply:
Yes, I'll add in the comment but for posterity, the context is with low resources (1 Gi) of memory the buffer was spent locking up the queue to load up to 100 MB of records. This meant that in certain situations 100 MB of memory could not be freed until the full flush, write/commit and return state was completed. Lowering this amount is one of the levers of improving speed by using a sliding window technique by breaking up records into smaller batches

It also leverages having more workers/threads being in use that would otherwise be idle since we fix the workers to 5 threads